templates/logrotate.j2

```diff
@@ -0,0 +1,49 @@
+# Use system user
+su {{ system_user }} {{ system_user }}
+
+# Create dedicated subdirectory for rotated files
+createolddir 770 {{ system_user }} {{ system_user }}
+
+# Frequency of logs rotation
+hourly
+maxage 7
+rotate 10800
+
+# Naming of rotated files should be in the format:
+dateext
+dateformat -%V_%H%M
+
+# Settings to prevent misconfigurations and unwanted behaviours
+ifempty
+missingok
+nocompress
+nomail
+nosharedscripts
+nocopytruncate
+
+{{ snap_common_directory }}/var/log/mysql/error.log {
+    olddir archive_error
+    postrotate
+        juju_command=/usr/bin/juju-run
+        if command -v /usr/bin/juju-exec; then juju_command=/usr/bin/juju-exec; fi
+        "$juju_command" -u {{ unit_name }} LOGS_TYPE=error JUJU_DISPATCH_PATH=hooks/flush_mysql_logs {{ charm_directory }}/dispatch
```

Comment:
me again with the enum idea...
```suggestion
        "$juju_command" -u {{ unit_name }} LOGS_TYPE="ERROR" JUJU_DISPATCH_PATH=hooks/flush_mysql_logs {{ charm_directory }}/dispatch
```
i.e. setting uppercase for this and the others

Reply:
addressed in 16a010bc74dc52227bfaceb1fe2040e7ad7a7b26